Update advanced_log_collection.md (#16239)
DataDog/documentation: content/en/agent/logs/advanced_log_collection.md

**Note**: The Agent requires read and execute permissions on a directory to list all the available files in it.

> **Public Beta Feature: Prioritize taling matched log files based on the last modification time**

Reviewer: ```suggestion
**Public Beta Feature: Prioritize tailing matched log files based on the last modification time**
```

> **Public Beta Feature: Prioritize taling matched log files based on the last modification time**

Reviewer 2: ```suggestion
## Tail most recently modified files first

**Note:** This feature is in public beta.
```

> The default behavior for the Datadog Agent to prioritize which file to tail first in the defined directory path is based on the file name, specifically in reverse lexicographical order. To have Agent tailing files based on the file modification time, you can set the configuration option `logs_config.file_wildcard_selection_mode` to be `by_modification_time`. 

Reviewer 2: ```suggestion
When prioritizing files to tail, the Datadog Agent sorts the filenames in the directory path by reverse lexicographic order. To sort files based on file modification time, set the configuration option `logs_config.file_wildcard_selection_mode` to the value `by_modification_time`.
```

> This will be especially helpful when the number of total log file match detected exceeds `logs_config.open_files_limit`. In this case, this option will ensure that the most recent updated files will be tailed first in the defined directory path. To change it back to the default behavor, you can set the configuration option `logs_config.file_wildcard_selection_mode` to be `by_name`. Please note that this feature will only be available for use for Agent version on 7.40.0 or above.

Reviewer 2: ```suggestion
This option is helpful when the number of total log file matches exceeds `logs_config.open_files_limit`. Using `by_modification_time` ensures that the most recently updated files are tailed first in the defined directory path.

To restore default behavior, set the configuration option `logs_config.file_wildcard_selection_mode` to the value`by_name`.

This feature requires Agent version 7.40.0 or above.
```

**Note**: The Agent requires read and execute permissions on a directory to list all the available files in it.

> **Public Beta Feature: Prioritize tailing matched log files based on the last modification time**

Reviewer 2: ```suggestion
## Tail most recently modified files first
**Note:** This feature is in public beta.
```
